### PR TITLE
Fixing the issue #2791 (Export backups to a sub-folder)

### PR DESCRIPTION
**Pull request by frankabjr**
Create a directory TextSecureBackup into the sdcard path, fixing the issue #2791.

### CODE REVIEW

File: src/org/thoughtcrime/securesms/database/PlaintextBackupExporter.java [hunk 1]
```diff
@@ -24,9 +24,13 @@ private static void verifyExternalStorageForPlaintextExport() throws NoExternalS
       throw new NoExternalStorageException();
   }
 
-  private static String getPlaintextExportDirectoryPath() {
+  public static String getPlaintextExportDirectoryPath() {
     File sdDirectory = Environment.getExternalStorageDirectory();
-    return sdDirectory.getAbsolutePath() + File.separator + "TextSecurePlaintextBackup.xml";
+    File backupDirectory = new File(String.format("%s%s%s", sdDirectory.getAbsolutePath(), File.separator, "TextSecureBackup"));
```
Comment (mcginty) [inline src/org/thoughtcrime/securesms/database/PlaintextBackupExporter.java]: `new File(String parent, String child);`

Comment (frankabjr) [inline src/org/thoughtcrime/securesms/database/PlaintextBackupExporter.java]: Ok. I'll adjust right now.

File: src/org/thoughtcrime/securesms/database/PlaintextBackupExporter.java [hunk 2]
```diff
@@ -24,9 +24,13 @@ private static void verifyExternalStorageForPlaintextExport() throws NoExternalS
       throw new NoExternalStorageException();
   }
 
-  private static String getPlaintextExportDirectoryPath() {
+  public static String getPlaintextExportDirectoryPath() {
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/database/PlaintextBackupExporter.java]: This should not be public

File: src/org/thoughtcrime/securesms/database/PlaintextBackupImporter.java [hunk 3]
```diff
@@ -36,8 +36,7 @@ private static void verifyExternalStorageForPlaintextImport() throws NoExternalS
   }
 
   private static String getPlaintextExportDirectoryPath() {
-    File sdDirectory = Environment.getExternalStorageDirectory();
-    return sdDirectory.getAbsolutePath() + File.separator + "TextSecurePlaintextBackup.xml";
+    return PlaintextBackupExporter.getPlaintextExportDirectoryPath();
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/database/PlaintextBackupImporter.java]: What about clients with existing backups in the old location?

Comment (frankabjr) [inline src/org/thoughtcrime/securesms/database/PlaintextBackupImporter.java]: Oh yeah. I will rewrite the import to support both paths.

**Comment by patzi** [comment]
Hi, wouldn't it be better to export the backup to ```/TextSecure/backup/TextSecurePlaintextBackup.xml``` So that other stuff i.e. exported media could be exported to ```/TextSecure/media```. So both in a subdirectory in ```/TextSecure``` thats the way I understood #2791.

**Comment by frankabjr** [comment]
@patzi, ok. But to do this, I'll have to make changes in other places in the source. I'll try do this.

**Comment by patzi** [comment]
Oh, I haven't thought that you should also work on exporting media to the subdirectory but I just thought it would be better to have the export in a subdirectory of ```/TextSecure``` so that later on media could also be exported there. You know just to have a better base for future development.

**Comment by frankabjr** [comment]
I did not test the Encrypted Backup because this functionality is not working in the current version of TS. But I resolved change this for keep the directory tree standard.

**Comment by frankabjr** [comment]
// FREEBIE

**Comment by f41c0r** [comment]
probably you should rebase into one commit, if you want this to get merged faster

**Comment by frankabjr** [comment]
Ok. I'll close this PR and make another with all changes in only one commit.

**Pull request closed without merging**